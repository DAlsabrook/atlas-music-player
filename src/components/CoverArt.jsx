import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css';

export function CoverArt(props) {
    return (
        <div className="w-full h-full">
            {props.loading ? (
                <Skeleton className="w-full h-90 rounded-lg" />
            ) : (
                <img src="/src/assets/placeholder.svg" alt="Cover art" className="w-full h-full rounded-lg" />
            )}
        </div>
    );
}
