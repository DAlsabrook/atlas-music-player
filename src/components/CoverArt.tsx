import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css';

export function CoverArt({loading}: {loading: Boolean}) {
    return (
        <div className="w-full h-full">
            {loading ? (
                <Skeleton className="w-full h-90 rounded-lg" />
            ) : (
                <img src="/src/assets/placeholder.svg" alt="Cover art" className="w-full h-full rounded-lg" />
            )}
        </div>
    );
}
