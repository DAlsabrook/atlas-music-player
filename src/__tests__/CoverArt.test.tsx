import { afterEach, expect, test } from "vitest";
import { cleanup, render } from "@testing-library/react";
import { CoverArt } from "@/components/CoverArt.tsx";

// Cleans DOM and any change that may have happened in a test
afterEach(() => {
    cleanup();
});

export type Song = {
  artist: String
  cover: string
  duration: Number
  genre: String
  id: String
  lyrics: String
  song: String
  title: String
}

const fetchPlaylist = async () => {
    const response = await fetch('http://localhost:5173/api/v1/playlist');
    if (response.ok) {
        const data = await response.json();
        const detailedSongs = await Promise.all(data.map(async (song: Song) => {
        const songDetailsResponse = await fetch(`http://localhost:5173/api/v1/songs/${song.id}`);
        if (songDetailsResponse.ok) {
            const songDetails = await songDetailsResponse.json();
            return { ...song, ...songDetails };
        }
        return song;
        }));
        const fullDetailedSongs = await Promise.all(detailedSongs.map(async (song: Song) => {
        const songDetailsResponse = await fetch(`http://localhost:5173/api/v1/lyrics/${song.id}`);
        if (songDetailsResponse.ok) {
            const songDetails = await songDetailsResponse.json();
            return { ...song, ...songDetails };
        }
            return song;
        }));
        return fullDetailedSongs
    }
};

test("renders loading skeleton when loading is true", async () => {
  const playlist = await fetchPlaylist();
  if (!playlist) throw new Error('Playlist is undefined');
  const { container } = render(<CoverArt loading={true} playlist={playlist} currentSong={0} />);
  expect(container).toMatchSnapshot();
});

test("renders cover image when loading is false", async () => {
  const playlist = await fetchPlaylist();
  if (!playlist) throw new Error('Playlist is undefined');
  const { container } = render(<CoverArt loading={false} playlist={playlist} currentSong={0} />);
  expect(container).toMatchSnapshot();
});

test("renders lyrics when loading is false and image is hovered", async () => {
  const playlist = await fetchPlaylist();
  if (!playlist) throw new Error('Playlist is undefined');
  const { container } = render(<CoverArt loading={false} playlist={playlist} currentSong={0} />);
  expect(container).toMatchSnapshot();
});
