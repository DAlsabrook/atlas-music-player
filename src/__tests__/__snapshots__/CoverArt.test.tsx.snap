// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders cover image when loading is false 1`] = `
<div>
  <div
    class="relative w-full h-full"
  >
    <img
      alt="Cover art"
      class="w-full h-full rounded-lg"
      src="https://placehold.co/400"
    />
    <div
      class="absolute inset-0 flex items-center justify-center bg-(--bg-color) bg-opacity-50 opacity-0 hover:opacity-100 transition-opacity duration-300 rounded-lg overflow-hidden"
    >
      <div
        class="text-(--secondary) p-4 overflow-y-auto max-h-full"
      >
        <p
          class="font-bold mb-2"
        >
          Lyrics
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders loading skeleton when loading is true 1`] = `
<div>
  <div
    class="relative w-full h-full"
  >
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton w-full h-90 rounded-lg"
      >
        â€Œ
      </span>
      <br />
    </span>
  </div>
</div>
`;

exports[`renders lyrics when loading is false and image is hovered 1`] = `
<div>
  <div
    class="relative w-full h-full"
  >
    <img
      alt="Cover art"
      class="w-full h-full rounded-lg"
      src="https://placehold.co/400"
    />
    <div
      class="absolute inset-0 flex items-center justify-center bg-(--bg-color) bg-opacity-50 opacity-0 hover:opacity-100 transition-opacity duration-300 rounded-lg overflow-hidden"
    >
      <div
        class="text-(--secondary) p-4 overflow-y-auto max-h-full"
      >
        <p
          class="font-bold mb-2"
        >
          Lyrics
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
    </div>
  </div>
</div>
`;
